molgenis-security/src/main/java/org/molgenis/security/audit/AuthenticationEventListener.java

```diff
@@ -0,0 +1,82 @@
+package org.molgenis.security.audit;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.molgenis.audit.AuditEventPublisher;
+import org.springframework.context.event.EventListener;
+import org.springframework.security.authentication.event.AbstractAuthenticationFailureEvent;
+import org.springframework.security.authentication.event.AuthenticationSuccessEvent;
+import org.springframework.security.authentication.event.LogoutSuccessEvent;
+import org.springframework.security.web.authentication.session.SessionFixationProtectionEvent;
+import org.springframework.security.web.authentication.switchuser.AuthenticationSwitchUserEvent;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthenticationEventListener {
+
+  public static final String AUTHENTICATION_FAILURE = "AUTHENTICATION_FAILURE";
+  public static final String AUTHENTICATION_SUCCESS = "AUTHENTICATION_SUCCESS";
+  public static final String AUTHENTICATION_SWITCH = "AUTHENTICATION_SWITCH";
+  public static final String LOGOUT_SUCCESS = "LOGOUT_SUCCESS";
+  public static final String SESSION_ID_CHANGE = "SESSION_ID_CHANGE";
+
+  private final AuditEventPublisher auditEventPublisher;
+
+  public AuthenticationEventListener(AuditEventPublisher auditEventPublisher) {
+    this.auditEventPublisher = requireNonNull(auditEventPublisher);
+  }
+
+  @EventListener
+  public void onAuthenticationFailureEvent(AbstractAuthenticationFailureEvent event) {
+    Map<String, Object> data = new HashMap<>();
+    data.put("type", event.getException().getClass().getName());
+    data.put("message", event.getException().getMessage());
+    if (event.getAuthentication().getDetails() != null) {
+      data.put("details", event.getAuthentication().getDetails());
+    }
+    auditEventPublisher.publish(event.getAuthentication().getName(), AUTHENTICATION_FAILURE, data);
```

Comment:
I was going to nag here that timestamp from the ApplicationEvent gets lost here but spring boot does it too so never mind that :)